doc/decap/subnet_decap_HLD.md

```diff
@@ -0,0 +1,194 @@
+
+### 5.2 Scalability Requirements
+
+| Component          | Expected Value |
+| ------------------ | -------------- |
+| Tunnel             | N/A            |
+| Tunnel Decap Terms | N/A            |
+
+## 6 Module Design
+
+To support the Netscan probing over VLAN subnet, SONiC needs to generate decapsulation rules to decapsulate IPinIP packets with DIP as any IP address in VLAN subnet. The decapsulation rules will be generated based on the configured VLAN subnet.
+In this design, we propose the subnet decap feature that has workflow to enable SONiC to add/remove those decapsulation rules based on the configured VLAN subnets on the T0 SONiC.
+
+### 6.1 Tunnel Specification
+
+The tunnels in this design will be generated with the following attributes:
+| Attribute       | Value                         | Note                             |
+| --------------- | ----------------------------- | -------------------------------- |
+| name            | IPINIP_VLAN or IPINIP_V6_VLAN | One IPv4 tunnel, one IPv6 tunnel |
+| tunnel type     | IPinIP                        |                                  |
+| decap ECN mode  | copy_from_outer or standard   |                                  |
+| decap TTL mode  | pipe                          |                                  |
+| decap DSCP mode | uniform                       |                                  |
+
+The decapsulation termination entry will be created with the following attributes:
+| Attribute       | Value                               | Note                                                                                                                     |
+| --------------- | ----------------------------------- | ------------------------------------------------------------------------------------------------------------------------ |
+| term entry type | MP2MP                               | multi-point to multi-point                                                                                               |
+| dest IP         | the VLAN subnet                     |                                                                                                                          |
+| dest IP mask    | the vlan subnet mask                |                                                                                                                          |
+| source IP       | Netscan privately-owned subnet      | IPinIP packets that have source IP in the subnet can be safely assumed to be Netscan traffic instead of customer traffic |
+| source IP mask  | Netscan privately-owned subnet mask |                                                                                                                          |
+
+### 6.2 DB Schema
+
+#### 6.2.1 CONFIG_DB
+
+```
+### SUBNET_DECAP
+    ; Stores subnet based decapsulation configurations
+    key                 = SUBNET_DECAP|subnet_type                      ; SUBNET_DECAP|vlan
```

Comment:
what is the significance of the `subnet_type` here?
With `subnet_type` being `vlan`, is it helping in programming HW table in anyway?